pulp_smash/tests/rpm/cli/test_environments.py

```diff
@@ -0,0 +1,70 @@
+# coding=utf-8
+"""Tests for RPM package environments."""
+from __future__ import unicode_literals
+
+import subprocess
+import unittest2
+
+from pulp_smash import cli, config, utils
+from pulp_smash.constants import (
+    RPM_ENV_DESC,
+    RPM_ENV_NAME,
+    RPM_FEED_URL,
+)
+
+
+class UploadPackageEnvTestCase(unittest2.TestCase):
+    """Test whether Pulp can upload package environments into a repository.
+
+    This test case covers `Pulp # 1003`_ and the corresponding Pulp Smash
+    issue `Pulp Smash #319`_.
+
+    1. Create and sync a repository.
+    2. Upload the environmrnt into this repo and check if there's any error.
+    """
+
+    @classmethod
+    def setUpClass(cls):
+        """Create and sync a repository."""
+        cls.cfg = config.get_config()
+        cls.repo_id = utils.uuid4()
+        cls.client = cli.Client(cls.cfg)
+        cls.client.run(
+            'pulp-admin rpm repo create --repo-id {} --feed {}'
+            .format(cls.repo_id, RPM_FEED_URL).split()
+        )
+        try:
+            cls.client.run(
+                'pulp-admin rpm repo sync run --repo-id {}'
+                .format(cls.repo_id).split()
+            )
+        except subprocess.CalledProcessError:
+            cls.client.run(
+                'pulp-admin rpm repo delete --repo-id {}'
+                .format(cls.repo_id).split()
+            )
+            raise
+
+    @classmethod
+    def tearDownClass(cls):
+        """Destroy the repository."""
+        cli.Client(cls.cfg).run(
+            'pulp-admin rpm repo delete --repo-id {}'
+            .format(cls.repo_id).split()
+        )
+
+    def test_upload_environment(self):
+        """Test if package environments can be uploaded."""
+        proc_upload = cli.Client(self.cfg).machine.session().run(
```

Comment:
> Can environments be deleted

IMO, the package environments are associated with a repository...since the uploading requires a specific repo-id. In that case, when deleting the repository in tearDownClass step, the package envs should also be deleted? At least it can no longer be looked up through `pulp-admin rpm repo content environment --repo-id <repo-id>`